src/main/resources/schema/attemptToAcquireSemaphoreLock.ddl.sql

```diff
@@ -3,9 +3,13 @@ BEGIN
 	DECLARE lockKeyExists VARCHAR(256);
 	DECLARE countOutstanding INT(4);
 	DECLARE newToken VARCHAR(256);
+	
+	/* ERROR CODE 3572 indicates NOWAIT for a lock was encountered.  For such cases null is returned. See: PLFM-5193 */
+	DECLARE EXIT HANDLER FOR 3572 SELECT NULL AS TOKEN;
+	
 	START TRANSACTION;

```

Comment:
can we remove the transaction at the dao now that we have transactions in the stored procedure.